cl/citations/tasks.py

```diff
@@ -88,120 +102,123 @@ def find_citations_and_parentheticals_for_opinion_by_pks(
+        opinion_clusters_to_update = OpinionCluster.objects.filter(
+            sub_opinions__pk__in=opinion_ids_to_update
+        )
+        opinion_clusters_to_update.update(
+            citation_count=F("citation_count") + 1
+        )
+
+        if index:
+            add_items_to_solr.delay(
+                opinion_clusters_to_update.values_list("pk", flat=True),
+                "search.OpinionCluster",
             )
-            if index:
-                add_items_to_solr.delay(
-                    opinion_clusters_to_update.values_list("pk", flat=True),
-                    "search.OpinionCluster",
+        # Nuke existing citations and parentheticals
+        OpinionsCited.objects.filter(citing_opinion_id=opinion.pk).delete()
+        Parenthetical.objects.filter(describing_opinion_id=opinion.pk).delete()
+
+        # Create the new ones.
+        OpinionsCited.objects.bulk_create(
+            [
+                OpinionsCited(
+                    citing_opinion_id=opinion.pk,
+                    cited_opinion_id=_opinion.pk,
+                    depth=len(_citations),
                 )
+                for _opinion, _citations in citation_resolutions.items()
+            ]
+        )
+        Parenthetical.objects.bulk_create(parentheticals)
 
-            # Nuke existing citations and parentheticals
-            OpinionsCited.objects.filter(citing_opinion_id=opinion.pk).delete()
-            Parenthetical.objects.filter(
-                describing_opinion_id=opinion.pk
-            ).delete()
-
-            # Create the new ones.
-            OpinionsCited.objects.bulk_create(
-                [
-                    OpinionsCited(
-                        citing_opinion_id=opinion.pk,
-                        cited_opinion_id=_opinion.pk,
-                        depth=len(_citations),
-                    )
-                    for _opinion, _citations in citation_resolutions.items()
-                ]
+        # Update parenthetical groups for clusters that we have added
+        # parentheticals for from this opinion
+        for cluster_id in clusters_to_update_par_groups_for:
+            create_parenthetical_groups(
+                OpinionCluster.objects.get(pk=cluster_id)
             )
 
-            Parenthetical.objects.bulk_create(parentheticals)
-
-            # Update parenthetical groups for clusters that we have added
-            # parentheticals for from this opinion
-            for cluster_id in clusters_to_update_par_groups_for:
-                create_parenthetical_groups(
-                    OpinionCluster.objects.get(pk=cluster_id)
-                )
-
             # Save all the changes to the citing opinion (send to solr later)
-            opinion.save(index=False)
+        opinion.save(index=False)
 
-    # If a Solr update was requested, do a single one at the end with all the
-    # pks of the passed opinions
-    if index:
-        add_items_to_solr.delay(opinion_pks, "search.Opinion")
```

Comment:
I'm not deep into this area of the code, but it's weird that this is getting removed. I'm guessing it's just duplicative of the solr command above, but are we entirely sure? Is it worth looking at the commit that added this and seeing if there's a clue about why this is here?

Reply:
Looks like that happened here: https://github.com/freelawproject/courtlistener/commit/c5eefc5dfb80de73de9923bf9272c2eeb7502b10#diff-4d550ad2f1f29ad62f61c3b21017f6c7908f5c6fd737f221b6dcd0fc22a0b351R175-R181

Reply:
I believe I have preserved this particular call: https://github.com/ss108/courtlistener/blob/docket-citations-fn/cl/citations/tasks.py#L113

Reply:
Yep, my bad. Thanks.